src/test/java/net/minecraftforge/debug/EventPistonDebug.java

```diff
@@ -0,0 +1,115 @@
+package net.minecraftforge.debug;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockPistonExtension;
+import net.minecraft.block.BlockPistonExtension.EnumPistonType;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.state.BlockPistonStructureHelper;
+import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.init.Blocks;
+import net.minecraft.item.ItemBlock;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraft.world.World;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.world.PistonEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.event.FMLInitializationEvent;
+import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.common.registry.GameRegistry;
+
+@Mod(modid = EventPistonDebug.MODID)
+public class EventPistonDebug
+{
+    public static final String MODID = "EventPistonDebug";
+    
+    private static Block shiftOnMove = new BlockShiftOnMove();
+    @Mod.EventHandler
+    public void preInit(FMLPreInitializationEvent event)
+    {
+        GameRegistry.register(shiftOnMove);
+        GameRegistry.register(new ItemBlock(shiftOnMove).setRegistryName(shiftOnMove.getRegistryName()));
+    }
+
+    @Mod.EventHandler
+    public void init(FMLInitializationEvent event)
+    {
+        MinecraftForge.EVENT_BUS.register(this);
+    }
+
+    @SubscribeEvent
+    public void onPistonExtend(PistonEvent.PistonExtendEvent event)
+    {
+        if (event.getWorld().isRemote)
+        {
+            BlockPistonStructureHelper pistonHelper =
+                    new BlockPistonStructureHelper(event.getWorld(), event.getPos(), event.getFacing(), true);
+            for (EntityPlayer player : event.getWorld().playerEntities)
```

Comment:
You can just send chat message to `Minecraft.getMinecraft().player` since this code block only executes on the client thread.